server/src/main/java/org/opensearch/indices/replication/RemoteStoreReplicationSource.java

```diff
@@ -47,15 +55,11 @@ public void getCheckpointMetadata(
         ReplicationCheckpoint checkpoint,
         ActionListener<CheckpointInfoResponse> listener
     ) {
-        FilterDirectory remoteStoreDirectory = (FilterDirectory) indexShard.remoteStore().directory();
-        FilterDirectory byteSizeCachingStoreDirectory = (FilterDirectory) remoteStoreDirectory.getDelegate();
-        RemoteSegmentStoreDirectory remoteDirectory = (RemoteSegmentStoreDirectory) byteSizeCachingStoreDirectory.getDelegate();
-
         Map<String, StoreFileMetadata> metadataMap;
         // TODO: Need to figure out a way to pass this information for segment metadata via remote store.
         final Version version = indexShard.getSegmentInfosSnapshot().get().getCommitLuceneVersion();
         try {
-            RemoteSegmentMetadata mdFile = remoteDirectory.readLatestMetadataFile();
+            RemoteSegmentMetadata mdFile = remoteDirectory.init();
```

Comment:
Why are we calling `init` here? If we only need to read metadata file, `readLatestMetadataFile` should be sufficient.

Reply:
If we don't call init does that fetch the latest metadata or a cached version?

Reply:
> If we don't call init does that fetch the latest metadata or a cached version?

It will fetch the latest metadata. Internally `init` also calls the same method to get the latest metadata. But as @dreamer-89 explained, init also updates uploaded segment files in `RemoteSegmentStoreDirectroy`. I just thought that it may not be required here. But if we want to access segment files as part of reading metadata file, we should call `init` only.

Reply:
@dreamer-89 We'd still need to call `init` when we're fetching the segments right? The list of segments tracked by `RemoteSegmentStoreDirectory` would be initialised/updated only if we call invoke.